### PR TITLE
Set up Modrinth automatic publishing

### PR DESCRIPTION
**Pull request by wafflecoffee**
Assuming `$MODRINTH_TOKEN` is set at the org level, this should just work :slightly_smiling_face: 

Tested [on staging](https://staging.modrinth.com/mod/freeeedom/version/1.7.4%2Bkotlin.1.6.21.local) to work well.

### CODE REVIEW

**Comment by jakobkmar** [comment]
I think the reason this hasn't been done before was because modmuss wanted to redo the publishing logic as a whole - but I still think this PR should be merged until that's done 👍

File: build.gradle.kts [hunk 1]
```diff
@@ -157,6 +159,34 @@ if (curse_api_key != null) {
...
+        gameVersions.set(
+            listOf(
+                "1.14", "1.14.1", "1.14.2", "1.14.3", "1.14.4",
+                "1.15", "1.15.1", "1.15.2",
+                "1.16", "1.16.1", "1.16.2", "1.16.3", "1.16.4", "1.16.5",
+                "1.17", "1.17.1",
+                "1.18", "1.18.1", "1.18.2",
+                // "1.19",
+            )
+        )
```
Comment (jakobkmar) [inline build.gradle.kts:176]: this should be abstracted to a global list and used for both the curseforge and the modrinth task

**Comment by modmuss50** [comment]
Yes, I want to move this to a groovy build script so its easier to maintain parity with the other projects.

I also want to setup the changelog generation that api and loader use. Ill leave this PR open as a reminder, but for now ill keep manually uploading them. (Please make sure to poke me on discord if (when) I forget)

File: build.gradle.kts [hunk 2]
```diff
@@ -157,6 +159,34 @@ if (curse_api_key != null) {
...
+
+        uploadFile.set(tasks.getByName<RemapJarTask>("remapJar"))
+
+        versionName.set("Fabric Language Kotlin $version")
+    }
+    project.afterEvaluate {
+        tasks.getByName<TaskModrinthUpload>("modrinth") {
+            dependsOn(tasks.getByName<RemapJarTask>("remapJar"))
+        }
+    }
```
Comment (jakobkmar) [inline build.gradle.kts:188]:
```suggestion
    tasks {
        named("modrinth") {
            dependsOn(remapJar)
        }
    }
```

**Comment by jakobkmar** [comment]
> Yes, I want to move this to a groovy build script so its easier to maintain parity with the other projects.

could this PR maybe be merged until that's done, since up-to-date fabric-language-kotlin on modrinth is quite helpful for third-party applications depending on the modrinth API - as well as some launchers needing it on modrinth

**Comment by modmuss50** [comment]
Done in groovy. Thanks.

**Pull request closed without merging**